app/Http/Controllers/Event/EventController.php

```diff
@@ -116,12 +118,24 @@ public function update(EventRequest $request, $id)
 	}
 
 	/**
-	 * Cancel an event.
-	 * @param int $id event id.
+	 * Organization cancel an event
 	 */
+	public function cancel($id)
+	{
+		if(auth()->guard('organization')->user()->id == Event::findorfail($id)->organization()->id)
+		{
+			$this->eventService->cancel($id);
```

Comment:
Pass the organization to the cancel function as this function is used by both web and API. Checking for the authenticated organization must be done in the service. The service must return true or false.